Propagate pppoe-server session offset to PPP interface name, thanks to Aleksey Shipilev

http://bugs.gentoo.org/204476

diff -Nru rp-pppoe-3.10.orig/src/pppoe-server.c rp-pppoe-3.10/src/pppoe-server.c
--- rp-pppoe-3.10.orig/src/pppoe-server.c	2008-06-30 16:00:43.000000000 +0200
+++ rp-pppoe-3.10/src/pppoe-server.c	2008-06-30 22:35:11.000000000 +0200
@@ -1751,7 +1751,7 @@ startPPPDUserMode(ClientSession *session)
     }
     if (PassUnitOptionToPPPD) {
 	argv[c++] = "unit";
-	sprintf(buffer, "%u", (unsigned int) (ntohs(session->sess) - 1 - SessOffset));
+	sprintf(buffer, "%u", (unsigned int) (ntohs(session->sess) - 1));
 	argv[c++] = buffer;
     }
     argv[c++] = NULL;
@@ -1834,7 +1834,7 @@ startPPPDLinuxKernelMode(ClientSession *session)
     argv[c++] = "default-asyncmap";
     if (PassUnitOptionToPPPD) {
 	argv[c++] = "unit";
-	sprintf(buffer, "%u", (unsigned int) (ntohs(session->sess) - 1 - SessOffset));
+	sprintf(buffer, "%u", (unsigned int) (ntohs(session->sess) - 1));
 	argv[c++] = buffer;
     }
     argv[c++] = NULL;
